src/fheroes2/battle/battle_dialogs.cpp

```diff
@@ -844,52 +844,54 @@ int Battle::Arena::DialogBattleHero( const HeroBase & hero, const bool buttons,
     int col = ( Color::NONE == hero.GetColor() ? 1 : Color::GetIndex( hero.GetColor() ) + 1 );
     fheroes2::Blit( fheroes2::AGG::GetICN( ICN::VIEWGEN, col ), display, pos_rt.x + 133, pos_rt.y + 36 );
 
-    fheroes2::Point tp( pos_rt.x, pos_rt.y );
-
     std::string str;
     Text text;
     text.Set( Font::SMALL );
     str = hero.isCaptain() ? _( "Captain of %{name}" ) : _( "%{name} the %{race}" );
     StringReplace( str, "%{name}", hero.GetName() );
     StringReplace( str, "%{race}", Race::String( hero.GetRace() ) );
     text.Set( str );
-    tp.x = pos_rt.x + ( pos_rt.width - text.w() ) / 2;
-    tp.y = pos_rt.y + 11;
+    fheroes2::Point tp{ pos_rt.x + ( pos_rt.width - text.w() ) / 2, pos_rt.y + 11 };
     text.Blit( tp.x, tp.y );
+
+    const fheroes2::Point statsTextOffset{ pos_rt.x + 148 - dialogShadow.x, pos_rt.y + 40 };
+    const int32_t maxStatsTextWidth{ 111 };
+    const int32_t statsTextRowHeight{ 11 };
+
     str = _( "Attack" ) + std::string( ": " ) + std::to_string( hero.GetAttack() );
     text.Set( str );
-    tp.x = pos_rt.x + 190 - text.w() / 2;
-    tp.y = pos_rt.y + 40;
+    tp.x = statsTextOffset.x + ( maxStatsTextWidth - text.w() ) / 2;
+    tp.y = statsTextOffset.y;
     text.Blit( tp.x, tp.y );
     str = _( "Defense" ) + std::string( ": " ) + std::to_string( hero.GetDefense() );
     text.Set( str );
-    tp.x = pos_rt.x + 190 - text.w() / 2;
-    tp.y = pos_rt.y + 51;
+    tp.x = statsTextOffset.x + ( maxStatsTextWidth - text.w() ) / 2;
+    tp.y = statsTextOffset.y + statsTextRowHeight;
```

Comment:
This and next `tp.y = ` lines can be simplified to:
```suggestion
    tp.y +=  statsTextRowHeight;
```